backend/backup_media.js

```diff
@@ -0,0 +1,155 @@
+}
+
+try { fs.mkdirSync(config.dir); } catch (e) {}
+if (!fs.existsSync(config.dbFile)) {
+  fs.openSync(config.dbFile, 'w+');
+  fs.appendFileSync(config.dbFile, 'item_type,item_id,media_type,media_src,media_id,img_width,img_height  ' + "\n");
+} else {
+  fs.openSync(config.dbFile, 'w+');
+}
+
+const connection = mysql.createConnection({
+  host: config.host,
+  user: config.user,
+  password: config.password,
+  database: config.database
+});
+
+const processBatch = async (offset) => {
+  const result = await connection.promise().query(
+    'select * from `content_source` where `source` like "%<img%" or `source` like "%<video%" limit ' + offset + ',' + BATCH_SIZE
+  );
+
+  if (!result[0] || !result[0].length) {
+    console.log('Done');
+    process.exit(0);
+  }
+
+  for (let i = 0; i < result[0].length; i++) {
+    let entry = result[0][i];
+    let itemType = entry.ref_type;
+    let itemId = entry.ref_id;
+    let entrySource = entry.source;
+    const mediaItems = entrySource.match(/<(img|source).+src=['"]([^'"]+)['"]/g);
+    if (!mediaItems) {
+      continue;
+    }
+    for (let ii = 0; ii < mediaItems.length; ii++) {
+      const check = mediaItems[ii].match(/<(img|source).+src=['"]([^'"]+)['"]/);
+      let mediaSrc = check[2];
+      let mediaType = check[1] === MEDIA_TYPE_IMG ? MEDIA_TYPE_IMG : MEDIA_TYPE_VIDEO;
+      let mediaExtensionMatch = mediaSrc.match(/\.[a-z]{2,}(?:\?.+)?$/);
+      let mediaExtension;
+      if (!mediaExtensionMatch) {
+        if (mediaType === MEDIA_TYPE_IMG) {
+          mediaExtension = '.jpg';
+        } else {
+          mediaExtension = '.mp4';
+        }
+      } else {
+        mediaExtension = mediaExtensionMatch[0].replace(/\?.+?$/, '');
+      }
+      let outputFilename = crypto.createHash('sha256').update(mediaSrc).digest('hex') + mediaExtension;
+      let outputPath = config.dir + '/' + outputFilename;
+      if (fs.existsSync(outputPath)) {
+        console.log(outputPath, 'is already there');
+        saveEntry(mediaType, outputPath, outputFilename, mediaSrc, itemType, itemId);
+        continue;
+      }
+      try {
+        console.log(entrySource);
+        console.log(`Downloading ${mediaSrc} to ${outputPath}`);
+        if (await download(mediaSrc, outputPath)) {
+          saveEntry(mediaType, outputPath, outputFilename, mediaSrc, itemType, itemId);
+          console.log('Downloaded: ', mediaSrc);
+        }
+      } catch (e) {
+        console.error('Failed to download: ', mediaSrc);
+      }
+    }
+  }
+  await sleep(SLEEP_BETWEEN_BATCHES_MS);
+  await processBatch(offset + BATCH_SIZE);
+}
+
+(async () => {
+  const result = await connection.promise().query(
+    'select count(*) as total from `content_source` where `source` like "%<img%" or `source` like "%video%"'
+  );
+  console.log('Content sources with media: ', result[0][0].total);
+  console.log('Start processing');
+  await processBatch(0);
+  process.exit(0);
```

Comment:
Instead of terminating process, please add the `finally` block and call `await conn.end();` there.